default/python/stub_generator/parse_docs.py

```diff
@@ -110,16 +111,37 @@ def parse_name(txt):
     return [x[0] for x in args], return_type
 
 
-def merge_args(arg_types, is_class):
+def merge_args(name, raw_arg_types, is_class):
+    """
+    Merge multiple set of arguments together.
+
+    Single argument set is used as is.
+    If we have two unique argument sets, and on of them is empty, use keywords.
+    In other cases log error and use first one.
+
+    :param str name:
+    :param list[tuple] raw_arg_types:
+    :param bool is_class:
+    :rtype: (list[str], list[(str, str)])
+    """
+    # If wrapper define functions that have same name, and same arguments but different return types,
```

Comment:
Is there an example of where this code is needed with our wrappers?  I am having a hard time seeing how it would not be a mistake for the wrapper to try specifying two function definitions with the same names and arguments but different return types.

Reply:
> I am having a hard time seeing how it would not be a mistake for the wrapper to try specifying two function definitions with the same names and arguments but different return types.

It is a mistake which compiles.   I did not ever think that this is possible.
https://github.com/freeorion/freeorion/pull/2292

Reply:
> It is a mistake which compiles. I did not ever think that this is possible. # 2292

That example you cite is not really an example of having different return types, it seems to me.  But I suppose it's close enough to show that the problem of semi-redundant declarations with different return types *could* happen.  And in case that did happen, then the error logging here would at least flag that in a log file, ok.